File: server/DeepDame/src/main/java/com/deepdame/dto/user/UserDto.java
package com.deepdame.dto.user;


import com.deepdame.dto.role.RoleDto;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.time.LocalDateTime;
import java.util.*;

@Builder
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class UserDto {
    @EqualsAndHashCode.Include
    private UUID id;

    @NotEmpty
    private String username;

    @NotEmpty
    @Email
    private String email;

    @NotNull
    private Boolean emailValidated;

    @NotNull
    private Boolean bannedFromChat;

    @NotNull
    private Boolean bannedFromApp;
    private Set<RoleDto> roles;

    @NotNull
    private LocalDateTime createdAt;
}

File: client/deepdame/lib/dtos/UserDto.dart
import 'package:deepdame/models/User.dart';

class UserDTO {
  final String? id;
  final String? username;
  final String? email;
  final bool emailValidated;
  final bool bannedFromChat;
  final bool bannedFromApp;

  const UserDTO({
    this.id,
    this.username,
    this.email,
    required this.emailValidated,
    required this.bannedFromChat,
    required this.bannedFromApp,
  });

  static UserDTO fromJson(Map<String, dynamic> decodedJson) {
    return UserDTO(
      id: decodedJson['id'],
      username: decodedJson['username'],
      email: decodedJson['email'],
      emailValidated: decodedJson['emailValidated'],
      bannedFromApp: decodedJson['bannedFromApp'],
      bannedFromChat: decodedJson['bannedFromChat'],
    );
  }

  User toUser() {
    return User(
      id: id,
      username: username,
      email: email,
      emailValidated: emailValidated,
      bannedFromChat: bannedFromChat,
      bannedFromApp: bannedFromApp,
    );
  }
}
